internal/table/client.go

```diff
@@ -348,59 +356,60 @@ func (c *Client) BulkUpsert(
 // If the request is smaller than the maximum size or contains fewer than two rows, it returns the original request.
-func chunkBulkUpsertRequest(req *Ydb_Table.BulkUpsertRequest, maxBytes int) ([]*Ydb_Table.BulkUpsertRequest, error) {
-	// not a row bulk upsert request -> ret original request
-	if req.Rows == nil || req.Rows.Value == nil {
-		return []*Ydb_Table.BulkUpsertRequest{req}, nil
-	}
-
+func chunkBulkUpsertRequest(
+	dst *[]*Ydb_Table.BulkUpsertRequest,
+	req *Ydb_Table.BulkUpsertRequest,
+	maxBytes int,
+) error {
 	reqSize := proto.Size(req)
 
 	// not exceed the maximum size -> ret original request
 	if reqSize <= maxBytes {
-		return []*Ydb_Table.BulkUpsertRequest{req}, nil
+		*dst = append(*dst, req)
+
+		return nil
+	}
+
+	// not a row bulk upsert request -> ret original request
+	if req.GetRows() == nil || req.GetRows().GetValue() == nil {
+		return fmt.Errorf("ydb: request size (%d bytes) exceeds maximum size (%d bytes) "+
+			" but cannot be chunked (only row-based bulk upserts support chunking)", reqSize, maxBytes)
 	}
 
-	n := len(req.Rows.Value.Items)
+	n := len(req.GetRows().GetValue().GetItems())
 	if n == 0 {
-		return []*Ydb_Table.BulkUpsertRequest{req}, nil
+		return nil
 	}
 
 	// we cannot split one item and one item is too big
 	if n == 1 {
-		err := fmt.Errorf("ydb: single row size (%d bytes) exceeds maximum request size (%d bytes) - row is too large to process", reqSize, maxBytes)
-		return nil, err
+		return fmt.Errorf("ydb: single row size (%d bytes) exceeds maximum request size (%d bytes) "+
+			"- row is too large to process", reqSize, maxBytes)
 	}
 
-	right := splitBulkUpsertRequestAt(req, n/2)
+	left, right := splitBulkUpsertRequestAt(req, n/2)
 
-	leftChunks, err := chunkBulkUpsertRequest(req, maxBytes)
-	if err != nil {
-		return nil, err
-	}
-
-	rightChunks, err := chunkBulkUpsertRequest(right, maxBytes)
-	if err != nil {
-		return nil, err
+	if err := chunkBulkUpsertRequest(dst, left, maxBytes); err != nil {
+		return err
 	}
 
-	return append(leftChunks, rightChunks...), nil
+	return chunkBulkUpsertRequest(dst, right, maxBytes)
 }
 
 // splitBulkUpsertRequestAt splits a BulkUpsertRequest into two parts at the specified position.
 // It creates a new request with items from the specified position to the end,
 // while modifying the original request to contain only items before the position.
 // Returns the newly created request with the split items.
-func splitBulkUpsertRequestAt(req *Ydb_Table.BulkUpsertRequest, pos int) *Ydb_Table.BulkUpsertRequest {
-	items := req.Rows.Value.Items // save original items
+func splitBulkUpsertRequestAt(req *Ydb_Table.BulkUpsertRequest, pos int) (_, _ *Ydb_Table.BulkUpsertRequest) {
+	items := req.GetRows().GetValue().GetItems() // save original items
 	req.Rows.Value.Items = nil
 
-	right := proto.Clone(req).(*Ydb_Table.BulkUpsertRequest)
+	right, _ := proto.Clone(req).(*Ydb_Table.BulkUpsertRequest)
```

Comment:
Is it error ignore? when the error can happened?

What about return error or panic (if error not possible in real case)?